### PR TITLE
Added RegexpHeaderCheck with parameters

### LINKED ISSUE #156: Extend the RegexpHeaderCheck to validate years
See https://github.com/openhab/static-code-analysis/pull/155#discussion_r134210900 

>  martinvw
> The only drawback is that it still does not allow us to validate years, which is a common failure, its was better than it was.

> svilenvul 
> I hope that we can easily extend the check and customize it in a next PR :-)

### PR DESCRIPTION
**Pull request by svilenvul**
Closes #156

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: src/main/java/org/openhab/tools/analysis/checkstyle/ParameterizedRegexpHeaderCheck.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class ParameterizedRegexpHeaderCheck extends AbstractHeaderCheck {
+    // Default header formats
+    public static final String DEFAULT_JAVADOC_START_COMMENT = "/**";
+    public static final String DEFAULT_XML_START_COMMENT = "<!--";
+
+    private static final String SEPARATOR = "\n";
+    private static final String MSG_MISMATCH = "Header line doesn`t match pattern {0}";
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/checkstyle/ParameterizedRegexpHeaderCheck.java]: Very minor: doesn't should imho not be written with a backtick?

File: src/main/java/org/openhab/tools/analysis/checkstyle/ParameterizedRegexpHeaderCheck.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    public void setHeaderFormat(String headerFormat) {
+        this.headerFormat = headerFormat;
+    }
+
+    public void setValues(String[] values) {
+        this.values = values;
+    }
+
+    /** The compiled regular expressions. */
+    private final List<Pattern> headerRegexps = new ArrayList<>();
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/checkstyle/ParameterizedRegexpHeaderCheck.java]: Imho all the fields should be before all the methods.

File: src/test/resources/checks/checkstyle/parameterizedRegexpHeaderCheckTest/MissingHeader.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+public class ValidHeaderJavaFile {
```
Comment (martinvw) [inline src/test/resources/checks/checkstyle/parameterizedRegexpHeaderCheckTest/MissingHeader.java]: Maybe name the class MissingHeader

File: src/test/resources/checks/checkstyle/parameterizedRegexpHeaderCheckTest/HeaderWithWrongYear.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (c) 2010-2015 by the respective copyright holders.
+ *
+ */
+ 
+public class ValidHeaderJavaFile {
```
Comment (martinvw) [inline src/test/resources/checks/checkstyle/parameterizedRegexpHeaderCheckTest/HeaderWithWrongYear.java]: Maybe adapt all classnames

**Comment by svilenvul** [comment]
Thanks @martinvw !

**Pull request merged**